server/raft.go

```diff
@@ -473,6 +474,28 @@ func (s *Server) reloadDebugRaftNodes() {
 	s.rnMu.RUnlock()
 }
 
+func (s *Server) shutdownRaftNodes() {
+	if s == nil {
+		return
+	}
+	var nodes []RaftNode
+	s.rnMu.RLock()
+	if len(s.raftNodes) > 0 {
+		s.Debugf("Shutting down all raft nodes")
+	}
+	for _, n := range s.raftNodes {
+		nodes = append(nodes, n)
+	}
+	s.rnMu.RUnlock()
+
+	for _, node := range nodes {
+		if node.Leader() {
+			node.StepDown()
```

Comment:
I guess that this is the right thing to do since you know this server is going away. The "bad" thing is that when one shutdown all servers, then this is likely to cause another server to become leader, etc.. but again, I believe this is the right thing to do in general.

Reply:
Yeah, I do see re-elects on shutdown/delete of consumers and streams too as you describe.

Iirc we dont yet have a API to permentantly evict a node, only step-down, we'd need that

Reply:
By default we want to assume this server going away is a singelton event and try to transfer leadership for any leader based nodes.

We can permanently evict a peer. Just remove JS from its config and do a config reload. JS will remap around it and that peer is considered gone.

So thought there was on ops side.

Reload config with JS disabled.
LDM

Reply:
If a machine just dies, power off, dead? We need to help things around a bit by manually killing it off?

Reply:
Yes I remember, I thought we had that already but I could be mistaken.

Reply:
JS can be enabled on a machine not specifically listed in the routes, and will be remembered ofcourse, so suspect we'll need a way to get rid of those that die unexpectedly